src/Plugin/Filter/TokenFilter.php

```diff
@@ -89,7 +103,15 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function process($text, $langcode) {
-    return new FilterProcessResult($this->token->replace($text), [], ['langcode' => $langcode]);
+    $data = [];
+
+    $entity = drupal_static('token_filter_entity', NULL);
```

Comment:
Why not using here the current route match service? the we iterate through parameters and when we find a ContentEntityInterface (bang!) we generate the `$data[]` for it. It's somehow like you did in tips:

```
foreach ($this->currentRouteMatch->getParameters() as $parameter) {
  if (!empty($parameter) && $parameter instanceof ContentEntityInterface) {
    /** @var \Drupal\Core\Entity\ContentEntityInterface $parameter */
    $token_type = $this->tokenEntityMapper->getTokenTypeForEntityType($parameter->getEntityTypeId());
    $data[$token_type] = $parameter;
  }
}
```

Of course, we need to inject also the service.